logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -319,16 +327,58 @@ private void newCheckpointedHeadpage(int pageNum) throws IOException {
         this.headPage.forceCheckpoint();
     }
 
+    /**
+     * Write a {@link Queueable} element to the queue. This will block until the write succeeds
+     * @param queueable
+     * @return the written sequence number
+     * @throws IOException
+     */
+    public long write(Queueable queueable) throws IOException {
+        return write(queueable, -1);
+    }
+
+    /**
+     * Write a {@link Queueable} element to the queue. This will block until the write succeeds or timeout occurs
+     * @param queueable
+     * @param timeoutMillis maximum time to block until returning
```

Comment:
nit: can you add `-1` means no timeout ?  (assuming that is correct)

Reply:
Good call! Yes I can.